addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/handler/KM200GatewayHandler.java

```diff
@@ -0,0 +1,710 @@
+        return name.replace("#", "/");
+    }
+
+    /**
+     * Translates a service path to a service name (Replaces / through #)
+     *
+     * @param name
+     */
+    public static String translatesPathToName(String path) {
+        return path.replace("/", "#");
+    }
+
+    /**
+     * Refreshes a channel
+     *
+     * @param channel
+     */
+    public void refreshChannel(Channel channel) {
+        GetSingleKM200Runnable runnable = new GetSingleKM200Runnable(this, getDevice(), comm, channel);
+        logger.debug("starting single runnable.");
+        scheduler.schedule(runnable, 0, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Updates bridges properties
+     *
+     */
+    private void updateBridgeProperties() {
+        List<String> propertyServices = new ArrayList<String>();
+        propertyServices.add(KM200ThingType.GATEWAY.getRootPath());
+        propertyServices.add(KM200ThingType.SYSTEM.getRootPath());
+        Map<String, String> bridgeProperties = editProperties();
+
+        for (KM200ThingType tType : KM200ThingType.values()) {
+            List<String> asProperties = tType.asBridgeProperties();
+            String rootPath = tType.getRootPath();
+            if (rootPath.isEmpty()) {
+                continue;
+            }
+            logger.debug("Add Property rootPath: {}", rootPath);
+            KM200CommObject serObj = getDevice().getServiceObject(rootPath);
+            for (String subKey : asProperties) {
+                if (serObj.serviceTreeMap.containsKey(subKey)) {
+                    String subKeyType = serObj.serviceTreeMap.get(subKey).getServiceType();
+                    if (subKeyType.compareTo("stringValue") != 0 && subKeyType.compareTo("floatValue") != 0) {
+                        continue;
+                    }
+                    if (bridgeProperties.containsKey(subKey)) {
+                        bridgeProperties.remove(subKey);
+                    }
+                    logger.debug("Add Property: {}  :{}", subKey,
+                            serObj.serviceTreeMap.get(subKey).getValue().toString());
+                    bridgeProperties.put(subKey, serObj.serviceTreeMap.get(subKey).getValue().toString());
+                }
+            }
+        }
+        updateProperties(bridgeProperties);
+    }
+
+    /**
+     * This function checks whether the service has a replacement parameter
+     *
+     * @param channel
+     * @param device
+     */
+    public static String checkParameterReplacement(Channel channel, KM200Device device) {
+        String service = translatesNameToPath(channel.getProperties().get("root"));
+        if (service.contains(SWITCH_PROGRAM_REPLACEMENT)) {
+            String currentService = translatesNameToPath(channel.getProperties().get(SWITCH_PROGRAM_CURRENT_PATH_NAME));
+            if (device.containsService(currentService)) {
+                if (device.getServiceObject(currentService).getServiceType().equals("stringValue")) {
```

Comment:
Position literals first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)